### PR TITLE
Hotfix/i18n testing as specs

### PR DESCRIPTION
**Pull request by ulferts**
Rewritten all the tests regarding languages as specs, removed test_unit tests. That should make it possible to deactivate them if necessary.

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage. <a href='http://hotfix-i18n-testing-as-specs.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Click to open.</a>

File: lib/redmine/i18n.rb [hunk 1]
```diff
@@ -110,8 +110,8 @@ def all_languages
     end
 
     def find_language(lang)
-      @@languages_lookup = valid_languages.inject({}) {|k, v| k[v.to_s.downcase] = v; k }
-      @@languages_lookup[lang.to_s.downcase]
+      languages_lookup = valid_languages.inject({}) {|k, v| k[v.to_s.downcase] = v; k }
```
Comment (hschink) [inline lib/redmine/i18n.rb]: That looks like an application for [`each_with_object`](http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-each_with_object)

File: spec/lib/redmine/i18n_spec.rb [hunk 2]
```diff
@@ -70,14 +70,83 @@ module OpenProject
...
+      it 'includes en' do
         expect(all_languages).to include(:en)
       end
+      it 'includes de' do
+        expect(all_languages).to include(:de)
+      end
 
       it 'should return no js language as they are duplicates of the rest of the other language' do
-        expect(all_languages.any?{ |l| /\Ajs-/.match(l.to_s) }).to be_false
+        expect(all_languages.any?{ |l| /\Ajs-/.match(l.to_s) }).to be_falsey
```
Comment (hschink) [inline spec/lib/redmine/i18n_spec.rb]: Obviously true. :wink:

File: lib/redmine/i18n.rb (hunk 1 above)
Comment (myabc) [inline lib/redmine/i18n.rb]: :+1: for @hschink's suggestion.

Comment (ulferts) [inline lib/redmine/i18n.rb]: @myabc, @hschink as the languages_lookup is no longer necessary (in fact it hasn't been used for a long time) I now implemented it in one line without variable assignment.

**Pull request merged**